src/main/scala/analysis/Cfg.scala

```diff
@@ -51,382 +196,930 @@ trait CfgNode:
-                  } else {
-                    val functionEntryNode = CfgFunctionEntryNode(data = d.target)
-                    val functionExitNode = CfgFunctionExitNode(data = d.target)
-                    funcEntryExit.addOne(d.target -> (functionEntryNode, functionExitNode))
-                  }
-                }
-              } else {
-                clonedFunctions.remove(d.target.name)
-              }
 
-              //nodePool.setLatestAdded(funcEntryExit(d.target)._2)
-              if (d.returnTarget.isDefined) {
-                if (processedBlocks.contains(d.returnTarget.get.label)) {
-                  otherLatestAdded.add(latestAdded.get)
-                } else {
-                  visitBlock(d.returnTarget.get.label)
-                }
-              } else {
-                otherLatestAdded.add(latestAdded.get)
-              }
-            case i: IndirectCall =>
-              val call = CfgCommandNode(data = i)
-              cfg.nodeToBlock += (call -> func.blocks.find(block => block.label == blockName).get)
-              cfg.addNode(call)
-              cfg.addEdge(lastAdded, call)
-              latestAdded = Some(call)
-              //print(s"Indirect call not supported yet ${i} ${func.name}\n")
-//              if (i.returnTarget.isDefined) {
-//                visitBlock(i.returnTarget.get.label)
-//              } else {
-//                cfg.addEdge(nodePool.getLatestAdded, functionExitNode)
-//              }
-              if (i.target.name.equals("R30")) {
-                cfg.addEdge(latestAdded.get, functionExitNode)
-              } else {
-                Logger.info(s"Cannot resolve indirect call ${i} ${func.name}")
-              }
-              if (i.returnTarget.isDefined) {
-                if (processedBlocks.contains(i.returnTarget.get.label)) {
-                  otherLatestAdded.add(latestAdded.get)
-                } else {
-                  visitBlock(i.returnTarget.get.label)
-                }
-              } else {
-                otherLatestAdded.add(latestAdded.get)
-              }
+        prevStmtNode
+      }
+
+      /** All blocks end with jump(s), whereas some also start with a jump (in the case of no statements). Add these to
+        * the CFG and visit their target blocks for processing.
+        *
+        * @param jmps
+        *   Jumps in the current block being processed
+        * @param prevNode
+        *   Either the previous statement in the block, or the previous block's end node (in the case that this block
+        *   contains no statements)
+        * @param cond
+        *   Jump from `prevNode` to this. `TrueLiteral` if `prevNode` is a statement, and any `Expr` if `prevNode` is a
+        *   jump.
+        * @param solitary
+        *   `True` if this block contains no statements, `False` otherwise
+        */
+      def visitJumps(jmps: List[Jump], prevNode: CfgNode, cond: Expr, solitary: Boolean): Unit = {
```

Comment:
why use List[Jump] when we are just converting an ArrayBuffer when this is called in visitBlock